File: src/test/java/com/github/kpavlov/jreactive8583/example/ClientReconnectIT.java
package com.github.kpavlov.jreactive8583.example;

import org.junit.Test;


public class ClientReconnectIT extends AbstractIT {

    @Test
    public void testClientReconnect() throws Exception {
        TestUtil.waitFor("server started", server::isStarted);
        TestUtil.waitFor("client connected", client::isConnected);
        server.shutdown();
        TestUtil.waitFor("client was disconnected", () -> (!client.isConnected()));
        Thread.sleep(7000);
        server.init();
        server.start();
        TestUtil.waitFor("server started", server::isStarted);
        TestUtil.waitFor("client connected", client::isConnected);
    }
}

File: src/test/java/com/github/kpavlov/jreactive8583/example/ClientReconnectIT.kt
package com.github.kpavlov.jreactive8583.example

import org.awaitility.Awaitility
import org.junit.jupiter.api.Test
import java.util.concurrent.TimeUnit

class ClientReconnectIT : AbstractIT() {

    @Test
    @Throws(Exception::class)
    fun clientShouldReconnectWhenConnectionLost() {
        //given
        Awaitility.await().alias("server started").until(server::isStarted)
        Awaitility.await().alias("client connected").until(client::isConnected)

        //when
        server.shutdown()
        Awaitility.await().alias("client was disconnected").until { !client.isConnected }

        //then
        TimeUnit.SECONDS.sleep(7)
        server.init()
        server.start()
        Awaitility.await().alias("server started").until(server::isStarted)
        Awaitility.await().alias("client connected").until(client::isConnected)
    }
}
